game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/MarkCasualties.java

```diff
@@ -0,0 +1,190 @@
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_PREFIX;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SELECT_CASUALTIES;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SELECT_FIRST_STRIKE_CASUALTIES;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SELECT_NAVAL_BOMBARDMENT_CASUALTIES;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.UNITS;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.Matches;
+import games.strategy.triplea.delegate.battle.BattleActions;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.battle.MustFightBattle;
+import games.strategy.triplea.delegate.battle.steps.BattleStep;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.triplea.java.ChangeOnNextMajorRelease;
+import org.triplea.java.Interruptibles;
+import org.triplea.java.RemoveOnNextMajorRelease;
+
+/** Step where the casualties are moved from ALIVE to CASUALTY */
+@RequiredArgsConstructor
+public class MarkCasualties implements BattleStep {
+
+  private static final long serialVersionUID = -3823676731273987167L;
+
+  private final BattleState battleState;
+
+  private final BattleActions battleActions;
+
+  /** The side of the firing player */
+  private final BattleState.Side side;
+
+  private final FiringGroup firingGroup;
+
+  private final FireRoundState fireRoundState;
+
+  @ChangeOnNextMajorRelease(
+      "returnFire is ALL for everything except NavalBombardment and old saves."
+          + "Rework so that returnFire isn't needed at all.")
+  private final MustFightBattle.ReturnFire returnFire;
+
+  @Override
+  public List<String> getNames() {
+    return List.of(getName());
+  }
+
+  private String getName() {
+    return battleState.getPlayer(side.getOpposite()).getName()
+        + REMOVE_PREFIX
+        // displaying UNITS makes the text feel redundant so hide it if that is the group name
+        + (firingGroup.getDisplayName().equals(UNITS)
+            ? CASUALTIES_WITHOUT_SPACE_SUFFIX
+            : firingGroup.getDisplayName() + CASUALTIES_SUFFIX);
+  }
+
+  @Override
+  public Order getOrder() {
+    return Order.REMOVE_FIRE_ROUND_CASUALTIES;
+  }
+
+  @Override
+  public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
+    if (!battleState.getStatus().isHeadless()) {
+      notifyCasualties(bridge);
+    }
+
+    battleActions.removeCasualties(
+        fireRoundState.getCasualties().getKilled(),
+        returnFire,
+        side.getOpposite() == DEFENSE,
+        bridge);
+
+    if (firingGroup.isSuicideOnHit()) {
```

Comment:
This is where the isSuicideOnHit check is being used.  You had wondered about that in the FiringGroup PR.